opal/class/opal_fifo.h

```diff
@@ -258,11 +271,13 @@ static inline opal_list_item_t *opal_fifo_pop_atomic (opal_fifo_t *fifo)
     if (ghost == next) {
         void *tmp = item;
 
-        if (!opal_atomic_compare_exchange_strong_ptr (&fifo->opal_fifo_tail.data.item, &tmp, ghost)) {
+        if (!opal_atomic_compare_exchange_strong_ptr (&fifo->opal_fifo_tail.data.item, &tmp, (void *) ghost)) {
             while (ghost == item->opal_list_next) {
```

Comment:
```
do {
    opal_atomic_rmb ();
} while (ghost != item->opal_list_next);
``` 
?

Reply:
Yeah, cleaner that way. Changing it now.